controllers/managedocs_controller.go

```diff
@@ -722,6 +731,39 @@ func (r *ManagedOCSReconciler) getDesiredConvergedStorageCluster() (*ocsv1.Stora
 	return sc, nil
 }
 
+func (r *ManagedOCSReconciler) getDesiredConsumerStorageCluster() (*ocsv1.StorageCluster, error) {
+
+	capacityAsString := r.addonParams[consumerRequestedCapacity]
+	onboardingTicket := r.addonParams[consumerOnboardingTicket]
+	storageProviderEndpoint := r.addonParams[consumerStorageProviderEndpoint]
+
+	requestedCapacity, err := resource.ParseQuantity(capacityAsString)
+	if err != nil {
+		return nil, fmt.Errorf("invalid storage cluster capacity value: %v", capacityAsString)
+	}
+
+	currentState := r.storageCluster.Spec.ExternalStorage
+	sc := templates.ConsumerStorageClusterTemplate.DeepCopy()
+	es := &sc.Spec.ExternalStorage
+
+	if currentState.OnboardingTicket != onboardingTicket {
+		es.OnboardingTicket = onboardingTicket
+	}
+
+	if currentState.StorageProviderEndpoint != storageProviderEndpoint {
+		es.StorageProviderEndpoint = storageProviderEndpoint
+	}
+
+	if currentState.RequestedCapacity == nil || currentState.RequestedCapacity.Cmp(requestedCapacity) != -1 {
```

Comment:
Where did we define that we do not allow downscale for consumer clusters? Consumer downscale is just a quota changes and there should be no reason to prevent it